tests/basic/pure_test.go

```diff
@@ -5002,3 +5004,135 @@ var _ = Describe("{RebootingNodesWhileFADAvolumeCreationInProgressUsingZones}",
+		}
+		podNameinSC := "Torpedo-Test" + Inst().InstanceID
+		PodNameinFA := RealmName + "::" + podNameinSC
+
+		stepLog := "Create A pod inside Realm"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			isPodExists, err := pureutils.IsPodExistsOnMgmtEndpoint(faClient, PodNameinFA)
+			log.FailOnError(err, fmt.Sprintf("Failed to check if pod [%v] exists ", PodNameinFA))
+			if !isPodExists {
+				_, err = pureutils.CreatePodinFA(faClient, PodNameinFA)
+				log.FailOnError(err, fmt.Sprintf("Failed to create pod [%v] ", PodNameinFA))
+				podCreatedinFA, err := pureutils.IsPodExistsOnMgmtEndpoint(faClient, PodNameinFA)
+				log.FailOnError(err, fmt.Sprintf("Failed to check if pod [%v] exists ", PodNameinFA))
+				if !podCreatedinFA {
+					log.FailOnError(fmt.Errorf("Pod [%v] is not created in FA", PodNameinFA), "is pod created in FA?")
+				}
+			}
+			log.InfoD("Pod [%v] created in FA", PodNameinFA)
+
+		})
+		stepLog = "Assign the pod name to pure_fa_pod_name in the storage class"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			customConfigAppName := skipTestIfNoRequiredCustomAppConfigFound()
+
+			// save the original custom app configs
+			origCustomAppConfigs = make(map[string]scheduler.AppConfig)
+			for appName, customAppConfig := range Inst().CustomAppConfig {
+				origCustomAppConfigs[appName] = customAppConfig
+			}
+
+			// update the custom app config with pod name
+			Inst().CustomAppConfig[customConfigAppName] = scheduler.AppConfig{
+				PureFaPodName: podNameinSC,
+			}
+
+			log.Infof("JustBeforeEach using Inst().CustomAppConfig = %v", Inst().CustomAppConfig)
+			err = Inst().S.RescanSpecs(Inst().SpecDir, Inst().V.String())
+			log.FailOnError(err, fmt.Sprintf("Failed to rescan specs from %s", Inst().SpecDir))
+
+			context, err := Inst().S.Schedule(testName, scheduler.ScheduleOptions{
+				AppKeys:            Inst().AppList,
+				StorageProvisioner: fmt.Sprintf("%v", portworx.PortworxCsi),
+				Namespace:          testName,
+			})
+			log.FailOnError(err, "Failed to schedule application of %v namespace", testName)
+			contexts = append(contexts, context...)
+			ValidateApplications(contexts)
+
+		})
+		stepLog = "Fetch the pod name from the volume and compare with pod name present in storage class"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for _, ctx := range contexts {
+				volumes, err := Inst().S.GetVolumes(ctx)
+				log.FailOnError(err, "Failed to get volumes")
+				for _, volume := range volumes {
+					inspectedVolume, err := Inst().V.InspectVolume(volume.ID)
+					log.FailOnError(err, "Failed to inspect volume")
+					PodName := inspectedVolume.Locator.VolumeLabels["pure_fa_pod_name"]
+					if PodName == "" {
+						dash.VerifyFatal(PodName == "", PodName != "", fmt.Sprintf("is pod name present in volume labels?"))
+					} else if PodName != podNameinSC {
+						dash.VerifyFatal(PodName != podNameinSC, PodName == podNameinSC, fmt.Sprintf("is pod name in volume same as pod name in storage class?"))
```

Comment:
just this one check is enough 
dash.VerifyFatal(PodName , podNameinSC, "verify pod name in volume same as pod name in storage class")